README.md

```diff
@@ -61,6 +61,12 @@ MY_WEIRD_ENV_VAR=123_ABC #required,format=\d{3}_\w{3}
 
 In the above example, `\d{3}_\w{3}` is converted to a regexp and the value is checked against it.
 
+## If you use Docker, read this
+
+Docker automatically reads `.env` files present in the project's root. What this means is that, if you use `dotenv_validator` in an app you run in a docker container, you might get exceptions or warnings about your variables being in the wrong format even though they're right. The reason is that Docker parses the `.env` file and sets the `ENV` variable, but it isn't as careful when dealing with comments present in the file and these might end up getting inserted into your environment variables, throwing off `dotenv_validator`.
```

Comment:
How about this version?

```suggestion
Docker automatically reads `.env` files present in the project's root. What this means is that, if you use `dotenv_validator` in an app you run in a docker container, you might get exceptions or warnings about your variables being in the wrong format even though they're right. The reason is that, when running `docker-compose up`, docker-compose parses the `.env` file before the Rails application starts. It reads each line as is with a really simple parser (no quotes, comments and trailing spaces handling).

Then, since docker-compose already set the environment variables, the Dotenv gem won't override them. It parses the file as we'd expect, but it won't change env variables that are already set.

For more information check this [page](https://docs.docker.com/compose/environment-variables/#set-environment-variables-in-containers) from their docs.
```